core/src/main/scala/org/apache/spark/deploy/raydp/RayAppMaster.scala

```diff
@@ -96,8 +96,14 @@ class RayAppMaster(host: String,
 
     override def receive: PartialFunction[Any, Unit] = {
       case RegisterApplication(appDescription: ApplicationDescription, driver: RpcEndpointRef) =>
+
         logInfo("Registering app " + appDescription.name)
-        val app = createApplication(appDescription, driver)
+        var javaOpts = appendActorClasspath(appDescription.command.javaOpts)
+        // append java Xmx
+        javaOpts = javaOpts :+ s"-Xmx${appDescription.memoryPerExecutorMB}M"
+        val newCommand = appDescription.command.withNewJavaOpts(javaOpts)
+        val updatedAppDesc = appDescription.withNewCommand(newCommand)
+        val app = createApplication(updatedAppDesc, driver)
```

Comment:
Is it possible to do it in scheduler where appDesc is created? Is this our problem or Ray's?